cranelift/codegen/src/isa/x64/inst/mod.rs

```diff
@@ -426,6 +433,11 @@ impl Inst {
         }
     }
 
+    pub(crate) fn xmm_cmp_rm_r(op: SseOpcode, src: RegMem, dst: Reg) -> Inst {
+        debug_assert!(dst.get_class() == RegClass::V128);
```

Comment:
Not sure it's already included in the commits you're basing your work from, but: latest commit adding Bitcast contains a new helper to check that the `RegMem` operand is a `V128` reg, if it's a reg: `assert_reg_type`. If you have it in your history, can you use it here please? Otherwise, can you add a TODO here to use it?

Reply:
I can not find this symbol in my tree. This patch needs to be rebased but it is likely really a child patch or two that needs rebasing. I have not rebased this because I am not sure how best to proceed.